File: app/src/main/java/mb00/android/codehub/ui/user/view/UserOverviewFragment.java
package mb00.android.codehub.ui.user.view;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import com.bumptech.glide.Glide;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import mb00.android.codehub.R;
import mb00.android.codehub.api.builder.RetrofitBuilder;
import mb00.android.codehub.api.model.User;
import mb00.android.codehub.api.service.GitHubService;
import mb00.android.codehub.data.BundleKeys;
import mb00.android.codehub.data.PreferenceKeys;
import mb00.android.codehub.databinding.FragmentUserOverviewBinding;
import mb00.android.codehub.logic.utils.DateParser;
import mb00.android.codehub.ui.base.view.BaseBindingFragment;
import mb00.android.codehub.ui.user.adapter.UserFragmentPagerAdapter;
import mb00.android.codehub.ui.user.viewmodel.UserOverviewViewModel;
import retrofit2.Retrofit;

/**
 * Fragment containing user overview; launched from {@link UserFragmentPagerAdapter}
 */

public class UserOverviewFragment extends BaseBindingFragment<FragmentUserOverviewBinding, UserOverviewViewModel> {

    private String authHeader;
    private String userLogin;

    @Override
    protected int layout() {
        return R.layout.fragment_user_overview;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SharedPreferences preferences = getActivity().getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE);
        authHeader = preferences.getString(PreferenceKeys.AUTH_HEADER, "");
        userLogin = getArguments().getString(BundleKeys.USER_NAME);
    }

    @Override
    public void onStart() {
        super.onStart();

        userOverViewCall(authHeader, userLogin);
    }

    private void userOverViewCall(String header, String user) {
        Retrofit retrofit = RetrofitBuilder.getInstance();
        GitHubService service = retrofit.create(GitHubService.class);

        service.getUserOverview(header, user)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(this::setUserInfo, Throwable::getMessage);
    }
    
    private void setUserInfo(User user) {
        Glide.with(getActivity()).load(user.getAvatarUrl()).into(getBinding().userOverviewAvatarImageView);
        getBinding().userOverviewNameTextView.setText(user.getName());

        if (user.getName() == null || user.getName().isEmpty()) {
            getBinding().userOverviewNameTextView.setVisibility(View.GONE);
        }
        getBinding().userOverviewLoginTextView.setText(user.getLogin());
        getBinding().userOverviewCompanyTextView.setText(user.getCompany());
        if (user.getCompany() == null || user.getCompany().isEmpty()) {
            getBinding().userOverviewCompanyTextView.setVisibility(View.GONE);
        }
        getBinding().userOverviewLocationTextView.setText(user.getLocation());
        if (user.getLocation() == null || user.getLocation().isEmpty()) {
            getBinding().userOverviewLocationTextView.setVisibility(View.GONE);
        }
        getBinding().userOverviewEmailTextView.setText(user.getEmail());
        if (user.getEmail() == null || user.getEmail().isEmpty()) {
            getBinding().userOverviewEmailTextView.setVisibility(View.GONE);
        }
        getBinding().userOverviewWebsiteTextView.setText(user.getWebsite());
        if (user.getWebsite() == null || user.getWebsite().isEmpty()) {
            getBinding().userOverviewWebsiteTextView.setVisibility(View.GONE);
        }
        String creationDate = DateParser.parseEnglish(user.getCreationDate());
        getBinding().userOverviewCreationDateTextView.setText(creationDate);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/view/UserOverviewFragment.kt
package mb00.android.codehub.ui.user.view

import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.bumptech.glide.Glide
import mb00.android.codehub.R
import mb00.android.codehub.api.model.User
import mb00.android.codehub.data.BundleKeys
import mb00.android.codehub.data.PreferenceKeys
import mb00.android.codehub.databinding.FragmentUserOverviewBinding
import mb00.android.codehub.logic.utils.DateParser
import mb00.android.codehub.ui.base.view.BaseBindingFragment
import mb00.android.codehub.ui.user.adapter.UserFragmentPagerAdapter
import mb00.android.codehub.ui.user.viewmodel.UserOverviewViewModel
import timber.log.Timber

/**
 * Fragment containing user overview; launched from [UserFragmentPagerAdapter]
 */

class UserOverviewFragment : BaseBindingFragment<FragmentUserOverviewBinding, UserOverviewViewModel>() {

    private lateinit var authHeader: String
    private lateinit var userLogin: String

    override fun layout(): Int {
        return R.layout.fragment_user_overview
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val preferences = activity?.getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE)
        authHeader = preferences?.getString(PreferenceKeys.AUTH_HEADER, "") ?: ""
        userLogin = arguments?.getString(BundleKeys.USER_NAME) ?: ""
    }

    override fun onStart() {
        super.onStart()

        userOverViewCall(authHeader, userLogin)
    }

    private fun userOverViewCall(header: String, user: String) {
        disposables.add(viewModel.loadUserOverview(header, user)
                .subscribe({ user ->
                    setUserInfo(user)
                }, { error -> Timber.e(error.message) })
        )
    }

    private fun setUserInfo(user: User) {
        // set in layout
        val toolbarTextView = activity?.findViewById<TextView>(R.id.user_toolbar_text_view)
        val creationDate = DateParser.parseEnglish(user.creationDate)

        toolbarTextView?.text = user.login
        Glide.with(activity!!).load(user.avatarUrl).into(binding.userOverviewAvatarImageView)

        binding.userOverviewNameTextView.text = user.name
        binding.userOverviewLoginTextView.text = user.login
        binding.userOverviewCompanyTextView.text = user.company
        binding.userOverviewLocationTextView.text = user.location
        binding.userOverviewEmailTextView.text = user.email
        binding.userOverviewWebsiteTextView.text = user.website
        binding.userOverviewCreationDateTextView.text = creationDate

        if (user.name.isNullOrEmpty()) {
            binding.userOverviewNameTextView.visibility = View.GONE
        }
        if (user.company.isNullOrEmpty()) {
            binding.userOverviewCompanyTextView.visibility = View.GONE
        }
        if (user.location.isNullOrEmpty()) {
            binding.userOverviewLocationTextView.visibility = View.GONE
        }
        if (user.email.isNullOrEmpty()) {
            binding.userOverviewEmailTextView.visibility = View.GONE
        }
        if (user.website.isNullOrEmpty()) {
            binding.userOverviewWebsiteTextView.visibility = View.GONE
        }
    }

}
